### ISSUE TITLE
Android 6.0 crashes when finding minSdkVersion from ApplicationInfo

### ISSUE THREAD

**Issue by matthewkrueger** [DreaminginCodeZH/MaterialRatingBar#55]
Hello - a crash has popped up in production coming from this library. Here's the stack trace:

> Fatal Exception: java.lang.NoSuchFieldError: No instance field minSdkVersion of type I in class Landroid/content/pm/ApplicationInfo; or its superclasses (declaration of 'android.content.pm.ApplicationInfo' appears in /system/framework/framework.jar)
>        at me.zhanghai.android.materialratingbar.MaterialRatingBar.logRatingBarTintWarning(MaterialRatingBar.java:302)
>        at me.zhanghai.android.materialratingbar.MaterialRatingBar.setProgressTintList(MaterialRatingBar.java:164)
>        at com.myapp.dialog.CourseRatingDialogFragment.updateRatingStarColor(CourseRatingDialogFragment.java:141)
>        at com.myapp.dialog.CourseRatingDialogFragment.onRatingChanged(CourseRatingDialogFragment.java:67)
>        at me.zhanghai.android.materialratingbar.MaterialRatingBar.setSecondaryProgress(MaterialRatingBar.java:613)
>        at android.widget.RatingBar.updateSecondaryProgress(RatingBar.java:275)
>        at android.widget.RatingBar.onProgressRefresh(RatingBar.java:255)
>        at android.widget.ProgressBar.doRefreshProgress(ProgressBar.java:1390)
>        at android.widget.ProgressBar.refreshProgress(ProgressBar.java:1406)
>        at android.widget.ProgressBar.setProgress(ProgressBar.java:1452)
>        at android.widget.AbsSeekBar.setProgress(AbsSeekBar.java:1273)
>        at android.widget.AbsSeekBar.trackTouchEvent(AbsSeekBar.java:1173)
>        at android.widget.AbsSeekBar.onTouchEvent(AbsSeekBar.java:1065)
>        at android.view.View.dispatchTouchEvent(View.java:9998)
>        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2841)
>        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2458)
>        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2841)
>        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2458)
>        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2841)
>        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2458)
>        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2841)
>        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2458)
>        at com.android.internal.policy.PhoneWindow$DecorView.superDispatchTouchEvent(PhoneWindow.java:2831)
>        at com.android.internal.policy.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1863)
>        at android.app.Dialog.dispatchTouchEvent(Dialog.java:982)
>        at com.myapp.dialog.AbstractDialogFragment$1.dispatchTouchEvent(AbstractDialogFragment.java:55)
>        at com.android.internal.policy.PhoneWindow$DecorView.dispatchTouchEvent(PhoneWindow.java:2792)
>        at android.view.View.dispatchPointerEvent(View.java:10233)
>        at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:5398)
>        at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:5234)
>        at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4674)
>        at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4727)
>        at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4693)
>        at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:4835)
>        at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4701)
>        at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:4892)
>        at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4674)
>        at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4727)
>        at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4693)
>        at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4701)
>        at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4674)
>        at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:7360)
>        at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:7238)
>        at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:7199)
>        at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:7470)
>        at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:185)
>        at android.os.MessageQueue.nativePollOnce(MessageQueue.java)
>        at android.os.MessageQueue.next(MessageQueue.java:323)
>        at android.os.Looper.loop(Looper.java:143)
>        at android.app.ActivityThread.main(ActivityThread.java:7224)
>        at java.lang.reflect.Method.invoke(Method.java)
>        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
>        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)

Here's the applicable code in MaterialRatingBar:

```
    private void logRatingBarTintWarning() {
        if (getContext().getApplicationInfo().minSdkVersion >= Build.VERSION_CODES.LOLLIPOP) {
            return;
        }
        Log.w(TAG, "Non-support version of tint method called, this is error-prone and will crash" +
                " below Lollipop if you are calling it as a method of RatingBar instead of" +
                " MaterialRatingBar");
    }
```

So far, the crash has happened 100% on Samsung devices that are running Android 6.0. Any advice is appreciated, I'll have to look for a new library or fork this one if there isn't a quick fix. Thanks!

**Comment by matthewkrueger** [comment]
I ended up finding a solution. I changed to use:

`setSupportProgressTintList`

instead of 

`setProgressTintList`

and that resolved the crash.

**Comment by zhanghai** [comment]
Thanks for the info. To my surprise, lint never warned me about `minSdkVersion` being added in API level 24. Will fix.

**Comment by zhanghai** [comment]
Released in [v1.3.2](https://github.com/DreaminginCodeZH/MaterialRatingBar/releases/tag/v1.3.2).
